installer/debian/algorand/preinst

```diff
@@ -4,27 +4,45 @@ set -e
 
 PKG_NAME=@PKG_NAME@
 
-if [ "$1" != install ]
+if [ "$1" = install ]
 then
-    exit 0
+    if dpkg-query --list 'algorand*' &> /dev/null
+    then
+        if PKG_INFO=$(dpkg-query --show --showformat='${Package} ${Status}\n' 'algorand*' | grep "install ok installed")
+        then
+            # Filter out `algorand-indexer` and `algorand-devtools` packages, they are allowed to be
+            # installed alongside other `algorand` packages.
+            INSTALLED_PKG=$(grep -v -e algorand-indexer -e algorand-devtools <<< "$PKG_INFO" | awk '{print $1}')
+
+            if [ -n "$INSTALLED_PKG" ]
+            then
+                echo -e "\nAlgorand does not currently support multi-distribution installations!\n\
+        To install this package, it is necessary to first remove the \`$INSTALLED_PKG\` package:\n\n\
+        sudo apt-get remove $INSTALLED_PKG\n\
+        sudo apt-get install $PKG_NAME\n"
+                exit 1
+            fi
+        fi
+    fi
 fi
 
-if dpkg-query --list 'algorand*' &> /dev/null
+if [ "$1" = upgrade ]
 then
-    if PKG_INFO=$(dpkg-query --show --showformat='${Package} ${Status}\n' 'algorand*' | grep "install ok installed")
+    if [ -f /var/lib/algorand/genesis.json ]
     then
-        # Filter out `algorand-indexer` and `algorand-devtools` packages, they are allowed to be
-        # installed alongside other `algorand` packages.
-        INSTALLED_PKG=$(grep -v -e algorand-indexer -e algorand-devtools <<< "$PKG_INFO" | awk '{print $1}')
+        ALGO_MD5=$(md5sum /var/lib/algorand/genesis.json | awk '{print $1}')
```

Comment:
since you're using pipes, consider using `set -o pipefail`

Reply:
and then, use `export BASHOPTS` to export that to the subshell.

Reply:
Will fix.